cdap-security-spi/src/main/java/co/cask/cdap/security/spi/authorization/Authorizer.java

```diff
@@ -80,4 +82,62 @@
    * @param entity the {@link EntityId} on which all {@link Action actions} are to be revoked
    */
   void revoke(EntityId entity);
+
+  /**
+   * Returns all the {@link Privilege} for the specified {@link Principal}.
+   *
+   * @param principal the {@link Principal} for which to return privileges
+   * @return a {@link Set} of {@link Privilege} for the specified principal
+   */
+  Set<Privilege> listPrivileges(Principal principal);
```

Comment:
We don't check principal to exists so is it for all other possibilities as when it can throw exceptions ? 
In that case should grant/revoke also throw exception ?